lib/isolator/adapters/background_jobs/active_job.rb

```diff
@@ -1,10 +1,19 @@
 # frozen_string_literal: true
 
-Isolator.isolate :active_job,
-  target: ActiveJob::Base,
-  method_name: :enqueue,
+adapter = Isolator.isolate :active_job,
   exception_class: Isolator::BackgroundJobError,
   details_message: ->(obj) {
     "#{obj.class.name}" \
     "#{obj.arguments.any? ? " (#{obj.arguments.join(", ")})" : ""}"
   }
+
+Isolator.adapters.active_job.ignore_if do |job|
+  config = job.class.try(:enqueue_after_transaction_commit)
+  config == :always || (config == :default &&
+    ActiveJob::Base.queue_adapter.try(:enqueue_after_transaction_commit?) == true)
+end
+
+ActiveJob::Base.around_enqueue do |job, block|
+  # check if we are even notifying before calling `caller`, which is well known to be slow
+  adapter.notify(caller, job, job) if adapter.notify?(job)
+end
```

Comment:
Why did we replace the `#enqueue` patch with the callback?

Reply:
I need to access the job object in the `ignore_if` block, and from my understanding there's no way to do that unless I pass it in as an arg so it gets picked up [here](https://github.com/palkan/isolator/blob/bc4651ae60ece1d401d6a5306086de7adb743b03/lib/isolator/adapters/base.rb#L26)?

Reply:
Ah, I see. Thanks for the explanation.

I think, we should update the `ignore_if`/`notify?` interface to include the current object along with the arguments. Or, we can introduce a new option for the time being (and merge them in the next major release). Let me think about it.

Reply:
No worries, let me know what you decide on. I thought about it too but didn't want to introduce a breaking change just for this (on my own accord).

Reply:
I ended up adding the `ignore_on` option here: https://github.com/palkan/isolator/commit/4174095496e5fc086690eb8ddcc83c51def828fb

Let's try using it here.

Reply:
Updated, works like a charm :)